internal/controller/ingress/service_controller.go

```diff
@@ -148,7 +148,20 @@ func (r *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, err
 	}
 
-	if !controllers.IsUpsert(svc) {
+	// If the service is being deleted, we need to clean up any resources that are owned by it
```

Comment:
whats an example of an owned resource for a service? Is this the tunnel and TCP edge that the controller creates as a result of a LoadBalancer type service?

Reply:
Ahh.....yeah its further up in here, but https://github.com/ngrok/kubernetes-ingress-controller/blob/f2fb79c2f3b16e0bca537b593669d96bc76dacbe/internal/controller/ingress/service_controller.go#L72-L76